examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -99,98 +113,205 @@ Status ChefFanControlManager::HandleStep(StepDirectionEnum aDirection, bool aWra
     return SpeedSetting::Set(mEndpoint, newSpeedSetting);
 }
 
-CHIP_ERROR ChefFanControlManager::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+void ChefFanControlManager::HandleFanControlAttributeChange(AttributeId attributeId, uint8_t type, uint16_t size, uint8_t * value)
 {
-    VerifyOrDie(aPath.mClusterId == FanControl::Id);
-    VerifyOrDie(aPath.mEndpointId == mEndpoint);
+    ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange");
+    switch (attributeId)
+    {
+    case FanControl::Attributes::PercentSetting::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  PercentSetting");
+        DataModel::Nullable<Percent> percentSetting =
+            NumericAttributeTraits<DataModel::Nullable<Percent>>::StorageToWorking(*value);
+
+        // If the PercentSetting was set to Null (likely the mode was set to kAuto).
+        // Update PercentCurrent to 0 here as it can't be nullable and we don't have a "real"
+        // value for the Fan percentage. The spec doesn't specify what to do in this case.
+        // If not Null, PercentCurrent tracks PercentSetting's value.
+        SetPercentCurrent(percentSetting.IsNull() ? 0 : percentSetting.Value());
+        break;
+    }
+
+    case FanControl::Attributes::SpeedSetting::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  SpeedSetting");
+
+        // DataModel::Nullable<uint8_t> speedSetting = static_cast<DataModel::Nullable<uint8_t>>(*value);
+        DataModel::Nullable<uint8_t> speedSetting = NumericAttributeTraits<DataModel::Nullable<uint8_t>>::StorageToWorking(*value);
+        SetSpeedCurrent(speedSetting.IsNull() ? 0 : speedSetting.Value());
+        // Determine if the speed change should also change the fan mode
+        FanControl::Attributes::FanMode::Set(mEndpoint, SpeedToFanMode(mSpeedCurrent));
+        break;
+    }
+
+    case FanControl::Attributes::FanMode::Id: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::HandleFanControlAttributeChange  FanMode");
+
+        FanControl::FanModeEnum fanMode = static_cast<FanControl::FanModeEnum>(*value);
+        FanModeWriteCallback(fanMode);
+        break;
+    }
+
+    default: {
+        break;
+    }
+    }
+}
 
-    switch (aPath.mAttributeId)
+FanControl::FanModeEnum ChefFanControlManager::SpeedToFanMode(uint8_t speed)
+{
+    if (speed == 0)
+    {
+        return FanControl::FanModeEnum::kOff;
+    }
+    else if (speed <= FAN_MODE_LOW_UPPER_BOUND)
```

Comment:
clang-tidy does not run on this, however if it would, it would complain and ask for `no else after return` (https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html)

Reply:
Good idea. Fixed, thanks!